pennylane/_qubit_device.py

```diff
@@ -1081,22 +1082,14 @@ def _statistics_new(self, circuit: QuantumScript, shot_range=None, bin_size=None
                     result = method(obs, shot_range=shot_range, bin_size=bin_size)
                 else:
                     result = self._measure(m, shot_range=shot_range, bin_size=bin_size)
+
             elif obs.return_type is not None:
                 raise qml.QuantumFunctionError(
                     f"Unsupported return type specified for observable {obs.name}"
                 )
 
             # 2. Post-process statistics results (if need be)
-            float_return_types = {
-                Expectation,
-                Variance,
-                Probability,
-                VnEntropy,
-                MutualInfo,
-                ShadowExpval,
-            }
-
-            if obs.return_type in float_return_types:
+            if isinstance(m, (_Expectation, _Variance, _Probability, _VnEntropy, _MutualInfo)):
```

Comment:
Why was `ShadowExpval` removed?

Reply:
I must've missed it. Added `_ShadowExpval`.